pkg/auth/db_metadata.go

```diff
@@ -0,0 +1,180 @@
+		return time.Time{}, err
+	}
+	return time.Parse(time.RFC3339, value)
+}
+
+func writeMetadata(tx db.Tx, items map[string]string) error {
+	for key, value := range items {
+		_, err := tx.Exec(`
+			INSERT INTO auth_installation_metadata (key_name, key_value)
+			VALUES ($1, $2)
+			ON CONFLICT (key_name) DO UPDATE set key_value = $2`,
+			key, value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *DBMetadataManager) UpdateSetupTimestamp(ctx context.Context, ts time.Time) error {
+	_, err := d.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
+		return nil, writeMetadata(tx, map[string]string{
+			SetupTimestampKeyName: ts.UTC().Format(time.RFC3339),
+		})
+	}, db.WithLogger(logging.Dummy()))
+	return err
+}
+
+func (d *DBMetadataManager) IsInitialized(ctx context.Context) (bool, error) {
+	setupTimestamp, err := d.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
+		return getSetupTimestamp(tx)
+	}, db.WithLogger(logging.Dummy()), db.ReadOnly())
+	if err != nil {
+		var pgErr *pgconn.PgError
+		if errors.Is(err, pgx.ErrNoRows) ||
+			(errors.As(err, &pgErr) && pgErr.Code == pgerrcode.UndefinedTable) {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	return !setupTimestamp.(time.Time).IsZero(), nil
+}
+
+func (d *DBMetadataManager) Write(ctx context.Context) (map[string]string, error) {
+	metadata := make(map[string]string)
+	metadata["lakefs_version"] = d.version
+	metadata["golang_version"] = runtime.Version()
+	metadata["architecture"] = runtime.GOARCH
+	metadata["os"] = runtime.GOOS
+	dbMeta, err := d.db.Metadata(ctx)
+	if err == nil {
+		for k, v := range dbMeta {
+			metadata[k] = v
+		}
+	}
+	_, err = d.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
+		// write metadata
+		err = writeMetadata(tx, metadata)
+		if err != nil {
+			return nil, err
+		}
+		// write installation id
+		d.installationID, err = insertOrGetInstallationID(tx, d.installationID)
+		if err == nil {
+			metadata[InstallationIDKeyName] = d.installationID
+		}
+
+		// get setup timestamp
+		setupTS, err := getSetupTimestamp(tx)
+		if err == nil {
+			metadata[SetupTimestampKeyName] = setupTS.UTC().Format(time.RFC3339)
+		}
+		return nil, nil
+	}, db.WithLogger(logging.Dummy()))
+	return metadata, err
+}
+
+func exportMetadata(ctx context.Context, d *pgxpool.Pool, je *json.Encoder) error {
+	type metadataEntry struct {
+		KeyName  string `db:"key_name"`
+		KeyValue string `db:"key_value"`
+	}
+
+	// Gathering DB metadata to exclude from export
+	db := db.NewPgxDatabase(d)
+	dbMeta, err := db.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+
+	rows, err := d.Query(ctx, "SELECT key_name, key_value from auth_installation_metadata")
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+	scanner := pgxscan.NewRowScanner(rows)
+	for rows.Next() {
+		entry := metadataEntry{}
+		err := scanner.Scan(&entry)
+		if err != nil {
+			return err
+		}
+		if _, exist := dbMeta[entry.KeyName]; exist {
+			// DB metadata - skip
```

Comment:
What's the harm of exporting it too?

Reply:
No harm. Just preserves some irrelevant (IMHO) data in our system.
When lakeFS starts it rewrites the metadata, updating existing keys, and so, if you upgraded postgress, the new version will be reflected by the metadata.
If you switch to another technology, this value will not be updated so you will have irrelevant/outdated metadata stored
WDYT?